### ISSUE TITLE
Refactor public API into modules

### ISSUE THREAD

**Issue by philpax** [rustformers/llm#221]
The `llm` and `llm-base` crates are getting very-top-level heavy (there are dozens of types on the main page). We should reshape it so that it's much clearer where the entrypoints to the library are, and move types into public modules.

**Comment by danforbes** [comment]
Is there anything I can do to help with this? Can you be a little bit more specific about the way you'd like to see these changes materialize?

**Comment by hhamud** [comment]
This should be left last before merging in any bug fixes or enhancements. Is there a list of must-haves before releasing 0.2?

**Comment by philpax** [comment]
Indeed, this has been waiting on things to calm down. I would say that we should address any remaining PRs, then get this done. GGUF support may or may not be in 0.2. Haven't decided yet.
